hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsCompressor.java

```diff
@@ -209,13 +209,13 @@ private void writeDescriptor(MetricDescriptor originalDescriptor) throws IOExcep
 
     private MetricDescriptor prepareDescriptor(MetricDescriptor descriptor) {
         final ProbeUnit unit = descriptor.unit();
-        if (unit == null || !unit.isNewUnit()) {
-            return descriptor;
-        } else {
+        if (unit != null && unit.isNewUnit()) {
             return descriptor
                     .copy()
                     .withTag("metric-unit", unit.name())
                     .withUnit(null);
+        } else {
+            return descriptor;
```

Comment:
I don't understand this change.

Reply:
It's exactly the same :) I inverted the condition temporarily and accidentally it was included in my push. I will revert this.